R-package/R/mlp.R

```diff
@@ -0,0 +1,76 @@
+#' Convenience interface for multiple layer perceptron
+#' 
+#' @param data the input matrix.
+#' @param label the training label.
+#' @param hidden_node a vector containing number of hidden nodes on each hidden layer as well as the output layer.
+#' @param out_node the number of nodes on the output layer.
+#' @param dropout a vector containing the dropout ratio from the input layer to the last hidden layer.
+#' @param activation either a single string or a vector containing the names of the activation functions.
+#' @param out_activation a single string containing the name of the output activation function.
+#' @param device whether train on cpu (default) or gpu.
+#' @param eval_metric the evaluation metric/
+#' @param ... other parameters passing to \code{mx.model.FeedForward.create}/
+#' 
+#' @examples
+#' 
+#' require(mlbench)
+#' data(Sonar, package="mlbench")
+#' Sonar[,61] = as.numeric(Sonar[,61])-1
+#' train.ind = c(1:50, 100:150)
+#' train.x = data.matrix(Sonar[train.ind, 1:60])
+#' train.y = Sonar[train.ind, 61]
+#' test.x = data.matrix(Sonar[-train.ind, 1:60])
+#' test.y = Sonar[-train.ind, 61]
+#' model = mx.mlp(train.x, train.y, hidden_node = 10, out_node = 2, out_activation = "softmax", 
+#'                learning.rate = 0.1)
+#' preds = predict(model, test.x)
+#' 
+#' @export
+mx.mlp <- function(data, label, hidden_node = 1, out_node, dropout = NULL, 
+                   activation = "tanh", out_activation = "softmax",
+                   device=mx.ctx.default(), ...) {
+  
+  m <- length(hidden_node)
+  if (!is.null(dropout)) {
+    if (length(dropout) == 1) {
```

Comment:
Autoencoder was indeed different enough. For normal mlp, dropout was usually only before the last layer